models/order.js

```diff
@@ -40,6 +40,15 @@ orderSchema.statics.addEntries = function(orderId, newEntries) {
   })
 }
 
+orderSchema.statics.updateStatus = function(orderId, status) {
+  return this.findOne({ id: orderId }).then((order) => {
```

Comment:
I think that was because I was calling `findOneAndUpdate` on a standalone `Entry`, not the ones embedded within the order. I think you need to do `this.findOneAndUpdate({ id: orderId }, { status }, { new: true }).then(...)` to pass the updated `order` to the `then`.